File: Others/DesignPattern/src/Command/Invoker.java
package Command;

public class Invoker {
	private Command command;
    
    public void setCommand(Command command) {
        this.command = command;
    }
    /**
     * 要求该命令执行这个请求。
     */
    public void execute() {
        command.execute();
    }
}

File: Design_Patterns/行为型模式/Command/CommandPattern01/CommandPattern01/Command/Invoker.m
//
//  Invoker.m
//  CommandPattern01
//
//  Created by suxinde on 2016/11/15.
//  Copyright © 2016年 com.su. All rights reserved.
//

#import "Invoker.h"
#import "Command.h"

@implementation Invoker

- (void)setCommand:(Command *)command
{
    _command = command;
}

- (void)action
{
    [self.command execute];
}

@end
